modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-impl/src/main/java/com/liferay/headless/commerce/admin/catalog/internal/resource/v1_0/AttachmentResourceImpl.java

```diff
@@ -398,6 +443,32 @@ private Attachment _addOrUpdateAttachment(
 		ServiceContext serviceContext = _serviceContextHelper.getServiceContext(
 			cpDefinition.getGroupId());
 
+		long attachmentFileEntryId = GetterUtil.getLong(
+			attachment.getFileEntryId());
+
+		String attachmentGroupExternalReferenceCode =
+			attachment.getGroupExternalReferenceCode();
+
+		if (attachmentFileEntryId == 0) {
```

Comment:
this looks like if the `attachmentFileEntryId ` is not 0 it doesn't get set to the attachmentFileEntry, can you please check?